core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala

```diff
@@ -216,6 +217,98 @@ class DynamicBrokerReconfigurationTest extends ZooKeeperTestHarness with SaslSet
     verifySslConfig("", invalidSslProperties, configDesc)
   }
 
+  @Test
+  def testUpdatesUsingConfigProvider(): Unit = {
+    val PollingIntervalVal = "${file:polling.interval:interval}"
+    val PollingIntervalUpdateVal = "${file:polling.interval:updinterval}"
+    val SslTruststoreTypeVal = "${file:ssl.truststore.type:storetype}"
+    val SslKeystorePasswordVal = "${file:ssl.keystore.password:password}"
+
+    def executeConfigCommand(props: Properties): Unit = {
+      val propsFile = TestUtils.tempFile()
+      val propsWriter = new FileWriter(propsFile)
+      try {
+        clientProps(SecurityProtocol.SSL).asScala.foreach {
+          case (k, v) => propsWriter.write(s"$k=$v\n")
+        }
+      } finally {
+        propsWriter.close()
+      }
+
+      servers.foreach { server =>
+        val args = Array("--bootstrap-server", TestUtils.bootstrapServers(servers, new ListenerName(SecureInternal)),
```

Comment:
nit: we have this logic duplicated in `private def alterSslKeystoreUsingConfigCommand`, perhaps a helper method could help